### ISSUE TITLE
Add an option to hide bots' posts

### ISSUE THREAD

**Issue by Sylvhem** [tootsuite/mastodon#8293]
I think it should be useful to add an option to the local and global timelines to not display posts made by bots (aka. users who have the "service" Actor type).

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by Xkeeper0** [comment]
Crossrefing this with #8347 !

**Comment by trwnh** [comment]
I think filtering someone out of home/lists separately is a different issue altogether. There's a use case in filtering stuff out of Home but not in a List, or filtering stuff in a List but not in Home...

...actually, I just realized that particular behavior is a subset of bringing back per-column filters. I'll tag the relevant issues later when I get to a computer...

**Comment by johanbove** [comment]
I'm totally for more granular control over bots posting in our Mastodon instances. I would like to see a general filter option to control (See https://github.com/tootsuite/mastodon/issues/8347#issuecomment-414824194) if I, as a Mastodonian interested in getting in touch with people, be able to turn off ALL bot posts. Most of the current Bot posts are essentially RSS feed reposts which is are flooding the federated timeline. If a Mastodonian had the ability to filter out all Bot users, the Federated timeline would become useful again, even on the biggests Mastodon instances.

**Comment by Cassolotl** [comment]
There is a pull request for this that was opened in July 2019, #11272, which was approved 12 days ago and is just waiting on being merged by Gargron.

**Comment by umonaca** [comment]
PR #11272 and PR #10718 are related to this issue. They implement in different ways and both has some kind of unfixed bugs or problems.

**Comment by matt-y** [comment]
@umonaca It seems both of those PRs have been closed by a bot, ironically. 

I'd really like to see this feature get added, however two competing pull requests for the same issue that both perform filtering different is not going to be helpful in doing so. I think some direction on the preferred approach from the project maintainers is needed here.

**Comment by Cassolotl** [comment]
>I think some direction on the preferred approach from the project maintainers is needed here.

CC @Gargron @ThibG ?

**Comment by Cassolotl** [comment]
Gargron [said](https://github.com/tootsuite/mastodon/pull/10718#issuecomment-491352228):

> Until there is a way for mods to mark an account as a bot forcibly, I don't think there is any benefit from blocking notifications from bots.

I don't understand why the "allow mods to mark accounts as bots" feature is required to come first. Mods can't sit there watching the timeline and checking every bot-looking account to see if it's a bot, so even with the mod setting it'll never be 100%. The "hide bots" setting would partially work, until the mod-setting was added, and then it would work a bit better.

> The annoying bots would simply not mark themselves as bots. The not-annoying ones usually only respond to user input.

There are a lot of bots in my federated timeline that are marked as bots and that just post every x hours. I have to mute them all individually.

I would like to be able to hide the polite bots that mark themselves as such, even if the setting doesn't affect rude bots that don't.

**So, this feature could be added right now and would benefit people immediately.**

**Comment by Cassolotl** [comment]
At least two PRs have been written for this and then closed. Is this feature something that you'd like to add, @Gargron? Can you provide some guidance as to how someone might go about it in a way that you'd like, so that someone can submit a PR that has a chance of being accepted?

**Comment by FinecoFinit** [comment]
This issue was started in 2018 and to this day there is no solution?

**Comment by drequivalent** [comment]
How is this still open

**Comment by Cassolotl** [comment]
@drequivalent Do you mean "why hasn't this clearly excellent feature been made yet" or "why hasn't this obviously rubbish feature request been closed yet?"

Anyway, there are lots of issues on this list that have been open for years.

**Comment by janmechtel** [comment]
I'm considering to work on this. But it would be my first contribution here. Is somebody still interested and would explore together?

**Comment by LarsFosdal** [comment]
Personally - as a user of Mastodon - being able to hide or channel bots into separate streams would be very welcome.

**Comment by mattinatux** [comment]
I would also appreciate this dev.

I found this issue by searching _mastodon show in lists hide in home_. I definitely would love to separate these contexts (from the [Dealing with Unwanted Content](https://docs.joinmastodon.org/user/moderating/) doc):

- Home timeline and lists = matching statuses will be removed from your home feed and lists

**Comment by ajnuss** [comment]
Here to add my name to the list of people who would like to see a way to hide bot accounts. It seems a logical addition since there is already a way to flag and identify the accounts, and there is a filter function in the app. Adding a checkbox to the Filters page to filter accounts flagged as Bot should be extremely easy to implement. I am in agreement with those perplexed why this hasn't been done since it was first reported in 2018.

**Comment by janmechtel** [comment]
This became a lot less relevant for me since I stopped using the federated timeline and just focus on following hashtags, but maybe that's particular to the hashtags I'm following. There are simply not that many bots?

**Comment by RasheedAZ** [comment]
Please implement this. Federated timelines are almost entirely dominated by bots.

**Comment by TicTac9k1** [comment]
Yeah for federated timelines bots are not something I want to see. Makes sense when you follow a particular bot, but when it comes to finding valuable new accounts they get buried under in the federated timeline.

**Comment by davidmegginson** [comment]
Bump. Blocking the #bot hashtag isn't enough, because bot accounts don't use it consistently. We really need a one-click option to mute all automated accounts. I don't want bots to be part of my fediverse experience.

**Comment by jschmidt** [comment]
I'd love to see this. Tired of muting dozens of bots on various timelines.

**Comment by kartikynwa** [comment]
Bots make federated timeline unusable. Please implement this feature.

**Comment by davidmegginson** [comment]
The same applies to following hashtags.  Some bots just flood them.

**Comment by JeffGrewer** [comment]
[This guy, Rob Shearer](https://v.cx/2024/02/solar-bot), has 201 nearly identical bots on mas.to. I asked him if the'd hashtag them to be filtered, but he just called me an "anonymous rando" and refused.

**Comment by davidmegginson** [comment]
If the moderator(s) of mas.to aren't willing to take action, then I think there's a case for fedi-blocking the whole site.  But that doesn't take away from the need to add this feature to Mastodon properly.

**Comment by m33m33** [comment]
+1 we need a checkbox to filter bots at the very least from public timeline, or per column or whatever

**Comment by jwcrawley** [comment]
Mastodon has been gathering data on self-reported bot accounts. It's absurd that we've requested a feature to ignore self-reported bot accounts since 2018, with no movement at all to this effect.

This should be a check-box in our profile. And for the small amount of effort, would greatly enable the federated/firehose page to be imminently useful.

Assertion: bot accounts get few to no engagements of any kind (follows, likes, retoots, comments). Removing this garbage helps the community.

**Comment by SjoerdV** [comment]
Why not remove the entire bot API from mastodon? Just invite everyone employing bots to go to one of the already enshittified platforms.

A town square has no need for overhead drones dumping shit on everyone.

In this scenario only people are important, automation is not imho

**Comment by m33m33** [comment]
> Why not remove the entire bot API from mastodon? 

Some bots are perceived as useful, not the hourly-capybara-bot kind, more like bots focused on one community needs with unlisted or local posts.

**Comment by SjoerdV** [comment]
> I think we just need to give users ultimate control. There should be four options: hide all bot posts, only show posts from bots in your allowlist, show posts from bots except those on your blocklist, and show all bot posts.

I agree if these were the options combined with making it impossible to fake a regular user on the other end. I just mean that every post added through an api/app key should be marked as being a bot automatically.

**Comment by trwnh** [comment]
There is no "bot API" -- *every* post goes through the same API. The "bot" in this case is referring to an optional checkbox you can enable on your own profile to signal that your account is automated. There's nothing to "fake".

**Comment by SjoerdV** [comment]
> There is no "bot API" -- *every* post goes through the same API. The "bot" in this case is referring to an optional checkbox you can enable on your own profile to signal that your account is automated. There's nothing to "fake".

I am sorry, but you clearly never played with the api and you need to create an app/api key in your account to do things automated. So this can be detected.

**Comment by trwnh** [comment]
i might have played with the API a bit while writing the documentation for it

anyways: every app has to go through the same process of registering itself and getting a key. this includes the official apps. even the web ui uses the api, although it gets its token from the session rather than from oauth.

**Comment by SjoerdV** [comment]
> i might have played with the API a bit while writing the documentation for it

Lol, Nice!

> anyways: every app has to go through the same process of registering itself and getting a key. this includes the official apps. even the web ui uses the api, although it gets its token from the session rather than from oauth.

I would be very fine to filter out anything oath btw, that way I can be really sure no automation is in play AND the web ui is fine
